packages/hash/frontend/src/pages/[account-slug]/types/new/entity-type.page.tsx

```diff
@@ -164,6 +196,24 @@ const Page: NextPageWithLayout = () => {
                 {...register("name", {
                   required: true,
                   disabled: isSubmitting,
+                  validate: async (value) => {
+                    if (!user.shortname) {
+                      throw new Error("User shortname must exist");
+                    }
+
+                    const entityTypeId = generateTypeId({
+                      domain: FRONTEND_URL,
+                      kind: "entity-type",
+                      title: value,
+                      namespace: user.shortname,
+                    });
```

Comment:
Non-blocking: this could be in a function so it could be reused here and inside `handleFormSubmit`.

Reply:
Done